metrics/web/lib/common/presentation/navigation/metrics_page/metrics_page_factory.dart

```diff
@@ -13,45 +15,68 @@ import 'package:metrics/project_groups/presentation/pages/project_group_page.dar
 /// A factory that is responsible for creating the [MetricsPage]
 /// depending on the [RouteConfiguration].
 class MetricsPageFactory {
-  /// Creates the [MetricsPage] from the given [RouteConfiguration].
+  /// Creates the [MetricsPage] using the given [routeConfiguration]
+  /// and [pageParameters].
   ///
-  /// If the given [RouteConfiguration] is `null` or contains the route name that
-  /// does not match to any of [RouteName]s returns the [DashboardPage].
-  MetricsPage create(RouteConfiguration configuration) {
-    final routeName = configuration?.name;
-    final routePath = configuration?.path;
+  /// If the given [routeConfiguration] is `null` or the [routeConfiguration]'s
+  /// name does not match to any of [RouteName]s returns the [DashboardPage].
+  MetricsPage create(
+    RouteConfiguration routeConfiguration,
+    PageParametersModel pageParameters,
+  ) {
+    Widget child = DashboardPage();
+    RouteName routeName = RouteName.dashboard;
+    String path = RouteConfiguration.dashboard().path;
 
-    switch (routeName) {
+    if (routeConfiguration == null) {
+      return _createMetricsPage(child, routeName, path, pageParameters);
+    }
+
+    final routeConfigurationName = routeConfiguration.name;
+
+    switch (routeConfigurationName) {
       case RouteName.loading:
-        return MetricsPage(
-          child: const LoadingPage(),
-          name: routePath,
-        );
+        child = const LoadingPage();
+        routeName = routeConfigurationName;
+        path = routeConfiguration.path;
```

Comment:
:todo:
These two lines repeat in every case of the switch. It seems like we can improve the code here to reduce duplication or remove it at all. DM me if you cannot find the best way.

I suspect, we can select the `child` only and use the `default` case to nullify the incorrect values.